baseplate/clients/thrift.py

```diff
@@ -216,20 +252,27 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                     result = method(*args, **kwargs)
                 except TTransportException as exc:
                     # the connection failed for some reason, retry if able
+                    success = "false"
+                    exception = exc
+
                     span.finish(exc_info=sys.exc_info())
                     last_error = str(exc)
                     if exc.inner is not None:
                         last_error += f" ({exc.inner})"
                     continue
-                except (TApplicationException, TProtocolException):
+                except (TApplicationException, TProtocolException) as exc:
                     # these are subclasses of TException but aren't ones that
                     # should be expected in the protocol. this is an error!
+                    success = "false"
+                    exception = exc
                     span.finish(exc_info=sys.exc_info())
                     raise
                 except Error as exc:
```

Comment:
are 4xx/client failures included in the above or this block?

Reply:
No they would be caught by the generic `TExeption` below it is my current understanding